lib/java_buildpack/framework/luna_security_provider.rb

```diff
@@ -144,20 +148,27 @@ def write_configuration(servers, groups)
 VirtualToken = {
 EOS
           groups.each_with_index { |group, index| write_group f, index, group }
-          write_epilogue f
+          write_epilogue f, groups
         end
       end
 
-      def write_epilogue(f)
+      def write_epilogue(f, groups)
         f.write <<EOS
 }
 
 HAConfiguration = {
   AutoReconnectInterval = 60;
   HAOnly = 1;
-  ReconnAtt = 20;
+  reconnAtt = -1;
```

Comment:
I don't have a strong opinion on how long, so long as it isn't unbounded.  Each service is unique and their retry characteristics are as well.  I mean, if you haven't been able to connect for 5 minutes, an hour, 6 hours, etc., how do HSM users typically decide that they're never going to be able to connect?  You might also take a look into what HSM behaviors you're trying to protect against; is it just intermittent network hiccups (then retry for a minute or two) or extended downtime due to system updates (then retry an hour or two).